pkg/flow/config_test.go

```diff
@@ -1,72 +1,49 @@
 package flow_test
 
 import (
-	"os"
 	"strings"
 	"testing"
 
 	"github.com/grafana/agent/pkg/flow"
-	"github.com/hashicorp/hcl/v2"
+	"github.com/grafana/agent/pkg/river/ast"
 	"github.com/stretchr/testify/require"
 
 	_ "github.com/grafana/agent/pkg/flow/internal/testcomponents" // Include test components
 )
 
 func TestReadFile(t *testing.T) {
 	content := `
-		testcomponents "tick" "ticker-a" {
+		testcomponents.tick "ticker_a" {
 			frequency = "1s"
 		}
 
-		testcomponents "passthrough" "static" {
+		testcomponents.passthrough "static" {
 			input = "hello, world!"
 		}
 	`
 
-	f, diags := flow.ReadFile(t.Name(), []byte(content))
+	f, err := flow.ReadFile(t.Name(), []byte(content))
+	require.NoError(t, err)
 	require.NotNil(t, f)
-	requireNoDiagErrors(t, f, diags)
 
 	require.Len(t, f.Components, 2)
-	require.Equal(t, "testcomponents.tick.ticker-a", getBlockID(f.Components[0]))
+	require.Equal(t, "testcomponents.tick.ticker_a", getBlockID(f.Components[0]))
 	require.Equal(t, "testcomponents.passthrough.static", getBlockID(f.Components[1]))
 }
 
 func TestReadFile_Defaults(t *testing.T) {
-	f, diags := flow.ReadFile(t.Name(), []byte(``))
+	f, err := flow.ReadFile(t.Name(), []byte(``))
 	require.NotNil(t, f)
-	requireNoDiagErrors(t, f, diags)
+	require.NoError(t, err)
 
 	require.Len(t, f.Components, 0)
 }
 
-func TestReadFile_InvalidComponent(t *testing.T) {
-	content := `
-		doesnotexist "hello-world" {
-		}
-	`
-
-	f, diags := flow.ReadFile(t.Name(), []byte(content))
-	require.Nil(t, f)
-	require.True(t, diags.HasErrors())
-	require.Equal(t, `Blocks of type "doesnotexist" are not expected here.`, diags[0].Detail)
-}
```

Comment:
This error was returned because of HCL's schema checks, which doesn't exist in River. Now, the controller will return an error for invalid components.